tests/jobscript/19-exit-script.t

```diff
@@ -19,31 +19,50 @@
 # Test exit-script.
 
 . "$(dirname "${0}")/test_header"
-set_test_number 9
+set_test_number 12
 
+# 1) Should run on normal successful job exit.
 install_suite "${TEST_NAME_BASE}" "${TEST_NAME_BASE}"
-
 run_ok "${TEST_NAME_BASE}-validate" cylc validate "${SUITE_NAME}"
-
-# 1) Should run on normal successful job exit.
-run_ok "${TEST_NAME_BASE}-run" \
+run_ok "${TEST_NAME_BASE}-success" \
   cylc run --debug --no-detach "${SUITE_NAME}" 
 grep_ok 'Cheesy peas!' "${SUITE_RUN_DIR}/log/job/1/foo/01/job.out"
 grep_fail 'Oops!' "${SUITE_RUN_DIR}/log/job/1/foo/01/job.err"
+purge_suite "${SUITE_NAME}"
 
 # 2) Should not run on internal early EXIT.
-run_fail "${TEST_NAME_BASE}-run" \
+install_suite "${TEST_NAME_BASE}" "${TEST_NAME_BASE}"
+run_ok "${TEST_NAME_BASE}-validate" \
+    cylc validate --set=EXIT=true "${SUITE_NAME}"
+run_fail "${TEST_NAME_BASE}-exit" \
   cylc run --debug --no-detach --set=EXIT=true "${SUITE_NAME}" 
 grep_fail 'Cheesy peas!' "${SUITE_RUN_DIR}/log/job/1/foo/01/job.out"
 grep_ok 'EXIT Oops!' "${SUITE_RUN_DIR}/log/job/1/foo/01/job.err"
+purge_suite "${SUITE_NAME}"
 
 # 3) Should not run on external job TERM.
-cylc run --set=NSLEEP=30 "${SUITE_NAME}"
-sleep 3
-CYLC_JOB_PID=$(sed -n 's/^CYLC_JOB_PID=//p' "${SUITE_RUN_DIR}/log/job/1/foo/01/job.status")
-kill -s TERM $CYLC_JOB_PID
-sleep 5
+install_suite "${TEST_NAME_BASE}" "${TEST_NAME_BASE}"
+run_ok "${TEST_NAME_BASE}-validate" \
+    cylc validate --set=NSLEEP=30 "${SUITE_NAME}"
+cylc run --no-detach --set=NSLEEP=30 "${SUITE_NAME}" \
+    <'/dev/null' 1>'/dev/null' 2>&1 &
```

Comment:
This intrigues me, why pass `/dev/null` to stdin?

Reply:
As discussed, this prevents the backgrounded command from getting polluted by STDIN. Not truly necessary here, but should make everything safer.